lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -335,17 +409,24 @@ def on_tls_conf_set(
             # write the admin cert conf on all units, in case there is a leader loss + cert renewal
             self.opensearch_config.set_admin_tls_conf(current_secrets)
 
-        if should_restart:
-            self.peers_data.put(Scope.UNIT, "must_reboot_node", True)
+        # In case of renewal of the unit transport layer cert - restart opensearch
+        if renewal and cert_type == CertType.UNIT_TRANSPORT:
+            self.on[self.service_manager.name].acquire_lock.emit(
+                callback_override="_restart_opensearch"
+            )
```

Comment:
will all the units restart, or only one?

Reply:
Yes, we have safeguards in `_start_opensearch`